### PR TITLE
Product Subscriptions: Shipping section for simple subscription product

### LINKED ISSUE #11242: Show shipping option in product form for simple subscription product
We should show the "Shipping" option under the "+ Add more details" of the product form for both simple subscription and variable subscription type products.

Right now, we show that setting only for variable subscription type.

| Simple subscription | Variable subscription |
|--------|--------|
| SimpleSubscriptionNoShipping | VariableSubscriptionShipping |

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #11242 

## Description
Show the "Shipping" option under the "+ Add more details" of the product form for a simple subscription type product.

## Testing instructions

**Prerequisite**
1. Install Woo Subscription plugin on your store
1. Create a subscriptions product from `wp-admin`

**Steps**
1. Launch the app and Log in to the store.
3. Navigate to Products tab > select "+" > select manual option > Subscription product.
4. Tap on `+ Add more details`
5. You should see the "Shipping" row
6. Tap on it and enter the shipping info
7. Save the product
8. Validate from the `wp-admin` page that the shipping info is properly saved.
9. From `wp-admin` page enable the "Virtual" or the "Downloadable" toggle

Screenshot 2023-11-22 at 11 35 58 AM

10. Open the product form in the mobile app and tap on `+ Add more details` 
11. Ensure that the "Shipping" row is not displayed for a virtual or downloadable product.

## Screenshots
| Simple subscription | Simple subscription virtual |
|--------|--------|
| Simulator Screenshot - iPhone 14 Pro - 2023-11-22 at 11 38 25 | Simulator Screenshot - iPhone 14 Pro - 2023-11-22 at 11 38 31 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift [hunk 1]
```diff
@@ -792,6 +795,32 @@ final class ProductFormActionsFactoryTests: XCTestCase {
         assertEqual(expectedBottomSheetActions, factory.bottomSheetActions())
     }
 
+    func test_view_model_for_subscription_product_when_product_is_virtual() {
```
Comment (hafizrahman) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift:798]: ❓ More of a question for my own learning: With this PR, we want to emphasize that a virtual subscription product should not have "Shipping: option in the bottom sheet action.

This unit test function is already correct, in the sense that shipping is not added in the "expectedBottomSheetsActions" part. But, there is nothing here that explicitly says we should not have "Shipping" action here. 

My question: is leaving things out here enough to communicate that requirement, or is it better to explicitly mention it (perhaps in a comment above the `let expectedBottomSheetActions: ` line below)?

Comment (hafizrahman) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift:798]: Or does it make sense to add something like:

```
let excludedBottomSheetActions: [ProductFormBottomSheetAction] = [.editShippingSettings]
XCTAssertFalse(factory.bottomSheetActions().contains(excludedBottomSheetActions))
```

**Review by hafizrahman — APPROVED** [review]
This works nicely in my test. I left a non-blocking comment, but otherwise it's good ❤️

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift:798]: Good observation, Hafiz! 🎉 

I agree that we can explicitly test for the "Shipping" section. Added a few more tests in https://github.com/woocommerce/woocommerce-ios/pull/11243/commits/24347e7d8461c6ed4d6b5351cf5dc69892482947

We can add more for variable subscription product type when we tackle https://github.com/woocommerce/woocommerce-ios/issues/11178

**Pull request merged**